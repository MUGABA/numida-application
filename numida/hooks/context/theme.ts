export type Theme = {
  background: string;
  primary: string;
  secondary: string;
  text: string;
  buttonText: string;
  border: string;
};

export const lightTheme: Theme = {
  background: "#ffffff",
  primary: "#30C2E3",
  secondary: "#D4FAF5",
  text: "",
  buttonText: "",
  border: "",
};

export const darkTheme: Theme = {
  background: "#121212",
  primary: "#bb86fc",
  secondary: "#03dac6",
  text: "#ffffff",
  buttonText: "#000000",
  border: "#333333",
};
