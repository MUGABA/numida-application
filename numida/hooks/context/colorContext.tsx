// ThemeContext.tsx
import React, { createContext, ReactNode, useContext, useState } from "react";
import { darkTheme, lightTheme, Theme } from "./theme";

interface ThemeContextProps {
  theme: Theme;
  toggleTheme: () => void;
}

const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);

export const ColorThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
}) => {
  const [theme, setTheme] = useState<Theme>(lightTheme);

  const toggleTheme = () => {
    setTheme((prevTheme) =>
      prevTheme === lightTheme ? darkTheme : lightTheme
    );
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useColorTheme = (): ThemeContextProps => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error("useTheme must be used within a ThemeProvider");
  }
  return context;
};
