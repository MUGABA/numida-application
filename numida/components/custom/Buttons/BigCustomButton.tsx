import { useColorTheme } from "@/hooks/context/colorContext";
import React from "react";
import { Pressable, StyleSheet, Text, View } from "react-native";

type Props = {
  buttonTitle: string;
  actionFuntion: () => void;
};

const BigCustomButton = ({ buttonTitle, actionFuntion }: Props) => {
  const { theme } = useColorTheme();
  return (
    <View>
      <Pressable
        style={{ ...styles.btn, backgroundColor: theme.primary }}
        onPress={async () => {
          await actionFuntion();
        }}
      >
        <Text style={styles.btnTitle}>{buttonTitle}</Text>
      </Pressable>
    </View>
  );
};

export default BigCustomButton;

const styles = StyleSheet.create({
  btn: {
    borderRadius: 15,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 80,
  },
  btnTitle: {
    paddingHorizontal: 8,
    color: "white",
    fontSize: 18,
    paddingVertical: 12,
    borderRadius: 5,
    textTransform: "uppercase",
  },
});
