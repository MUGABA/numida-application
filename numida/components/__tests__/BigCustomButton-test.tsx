import { ColorThemeProvider } from "@/hooks/context/colorContext";
import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import BigCustomButton from "../custom/Buttons/BigCustomButton";

const renderWithProvider = (ui: React.ReactElement) => {
  return render(<ColorThemeProvider>{ui}</ColorThemeProvider>);
};

describe("MyComponent", () => {
  it("It renders button title collectly", () => {
    const { getByText } = renderWithProvider(
      <BigCustomButton
        buttonTitle="Apply for a loan"
        actionFuntion={() => {}}
      />
    );
    expect(getByText("Apply for a loan")).toBeTruthy();
  });

  it("It works collectly of a button is clicked", () => {
    const onClickMock = jest.fn();
    const { getByText } = renderWithProvider(
      <BigCustomButton
        buttonTitle="Apply for a loan"
        actionFuntion={onClickMock}
      />
    );

    const button = getByText("Apply for a loan");

    fireEvent.press(button);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it(`renders correctly`, () => {
    const onClickMock = jest.fn();
    const { toJSON } = render(
      <ColorThemeProvider>
        <BigCustomButton
          buttonTitle="Submit"
          actionFuntion={onClickMock}
        ></BigCustomButton>
      </ColorThemeProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
