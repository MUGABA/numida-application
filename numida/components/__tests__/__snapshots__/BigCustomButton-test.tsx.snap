// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent renders correctly 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#30C2E3",
        "borderRadius": 15,
        "justifyContent": "center",
        "marginTop": 80,
      }
    }
  >
    <Text
      style={
        {
          "borderRadius": 5,
          "color": "white",
          "fontSize": 18,
          "paddingHorizontal": 8,
          "paddingVertical": 12,
          "textTransform": "uppercase",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
